### PR TITLE
[GNA] Move to GNA Lib 3.0.0.1455 with extended CNN2D capabilities for GNA HW 3.0

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Switch to GNA Lib 3.0.0.1455
 - Relax limitations for GNA 3.0 2D CNN in the validator code
 - Add new tests (positive and negative) for GNA 3.0 2D CNN
 - Remove outdated negative tests for GNA 3.0 2D CNN

### Tickets:
 - 72506

### CODE REVIEW

File: cmake/dependencies.cmake [hunk 1]
```diff
@@ -287,8 +287,8 @@ if(ENABLE_GNA)
             set(GNA_HASH "cc954e67525006bf8bd353a6682e38bf208f6d74e973e0fc292850e721f17452")
         endif()
         if(GNA_LIBRARY_VERSION STREQUAL "GNA2")
-            set(GNA_VERSION "03.00.00.1377")
-            set(GNA_HASH "d45fb48994d8c2803a16e88e29ae48851066325b97c1c6c4a5bf4f4573d55c65")
+            set(GNA_VERSION "03.00.00.1455")
+            set(GNA_HASH "8ac1af18eb32777b00193f4f8c252ee4f8bd64a9069138b4a5aaeebd82ead464")
```
Comment (kbruniec) [inline cmake/dependencies.cmake:291]:
```suggestion
            set(GNA_HASH "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855")
```

File: cmake/dependencies.cmake [hunk 2]
```diff
@@ -287,8 +287,8 @@ if(ENABLE_GNA)
             set(GNA_HASH "cc954e67525006bf8bd353a6682e38bf208f6d74e973e0fc292850e721f17452")
         endif()
         if(GNA_LIBRARY_VERSION STREQUAL "GNA2")
-            set(GNA_VERSION "03.00.00.1377")
-            set(GNA_HASH "d45fb48994d8c2803a16e88e29ae48851066325b97c1c6c4a5bf4f4573d55c65")
+            set(GNA_VERSION "03.00.00.1455")
+            set(GNA_HASH "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855")
```
Comment (kbruniec) [inline cmake/dependencies.cmake:291]:
```suggestion
            set(GNA_HASH "8ac1af18eb32777b00193f4f8c252ee4f8bd64a9069138b4a5aaeebd82ead464")
```

File: cmake/dependencies.cmake [hunk 3]
```diff
@@ -298,12 +298,14 @@ if(ENABLE_GNA)
             LIST(APPEND FILES_TO_EXTRACT_LIST gna_${GNA_VERSION}/linux)
         endif()
      
+        set(IE_PATH_TO_DEPS "http://nnt-srv01.inn.intel.com/dl_score_engine")
```
Comment (kbruniec) [inline cmake/dependencies.cmake:301]:
```suggestion
```

File: cmake/dependencies.cmake [hunk 4]
```diff
@@ -298,12 +298,14 @@ if(ENABLE_GNA)
...
         RESOLVE_DEPENDENCY(GNA
                 ARCHIVE_UNIFIED "GNA/GNA_${GNA_VERSION}.zip"
                 TARGET_PATH "${TEMP}/gna_${GNA_VERSION}"
                 VERSION_REGEX ".*_([0-9]+.[0-9]+.[0-9]+.[0-9]+).*"
                 FILES_TO_EXTRACT FILES_TO_EXTRACT_LIST
                 SHA256 ${GNA_HASH})
+        unset(IE_PATH_TO_DEPS)
```
Comment (kbruniec) [inline cmake/dependencies.cmake:308]:
```suggestion
```

**Review by sirzabek — APPROVED** [review]

File: src/plugins/intel_gna/backend/gna_limitations.hpp [hunk 5]
```diff
@@ -99,10 +104,19 @@ class Validator {
     RangeMultipleLimit inputChannelsNumberLimit{ {8, 384, "number of input channels"}, 8 };
 
     RangeMultipleLimit kernelNumberLimit{ {8, 256, "number of kernels"}, 8 };
```
Comment (krzysztofjozwiak) [inline src/plugins/intel_gna/backend/gna_limitations.hpp:106]: Need to be updated to 1024.

Comment (kbruniec) [reply src/plugins/intel_gna/backend/gna_limitations.hpp]: fixed

**Review by krzysztofjozwiak — APPROVED** [review]

**Pull request merged**